app/User.php

```diff
@@ -59,7 +60,27 @@ public function printJobs()
         return $this->hasMany('App\PrintJob');
     }
 
+    public function roles() {
+        return $this->belongsToMany(Role::class, 'role_users')->withPivot('object_id');;
+    }
+
+    public function hasAnyRole(array $roleNames, $objectId = null)
+    {
+        return $this->roles->contains(function($value, $key) use($roleNames, $objectId) {
+            return in_array($value->name, $roleNames) && $value->pivot->object_id == $objectId;
```

Comment:
I think for comparing *nullable* objects we should use `===`.
The value `0` equals to `null` using `==` operator, and would be different using `===`.
You can check it [here](https://repl.it/repls/ExaltedPettyFan).